presto-main/src/test/java/com/facebook/presto/operator/TestHashJoinOperator.java

```diff
@@ -859,9 +864,26 @@ private static LookupSourceFactory buildHash(boolean parallelBuild, TaskContext
         }
         getFutureValue(lookupSourceProvider).close();
 
+        for (Driver buildDriver : buildDrivers) {
```

Comment:
It is a bit confusing why we run `process` until `lookupSourceProvider.isDone()` above and then we run again here... if they can't be merged, can you comment here why we need the second process loop (and why it is backgrounded)